planner/core/rule_predicate_push_down.go

```diff
@@ -72,11 +72,11 @@ func splitSetGetVarFunc(filters []expression.Expression) ([]expression.Expressio
 
 // PredicatePushDown implements LogicalPlan PredicatePushDown interface.
 func (p *LogicalSelection) PredicatePushDown(predicates []expression.Expression) ([]expression.Expression, LogicalPlan) {
+	p.Conditions = expression.PropagateConstant(p.ctx, p.Conditions)
```

Comment:
Why this change can take effect?
I just see that there's `PropagateConstant` in Sel's `PredicatePushDown` and in `addSelection`.

Reply:
I had do some change, what I want to do is do `PropagateConstant`  before `PredicatePushDown` .